php-classes/Google/API.php

```diff
@@ -141,17 +151,39 @@ public static function executeBatchRequest(array $Requests)
 
                 // skip unparsable response bodies
                 if (count($responseParts) >= 2) {
-                    $responseBody = $responseParts[count($responseParts) - 2];
+                    $responseBody = json_decode($responseParts[count($responseParts) - 2], true);
+
+                    if (
+                        (
+                            $retry === true ||
+                            is_numeric($retry) &&
+                            (int)$retry <= static::$maxRetries
+                        ) &&
+                        isset($responseBody['error']) &&
+                        is_array($responseBody['error']) &&
+                        isset($responseBody['error']['code']) &&
+                        in_array($responseBody['error']['code'], static::$retryResponseCodes)
+                    ) {
+                        $failedRequests[$contentId] = $Requests[$contentId];
+                        $failedResponses[$contentId] = $responseBody['error'];
+                    }
+
                 } else {
                     continue;
                 }
 
-                $results[$contentId] = json_decode($responseBody, true);
+                $results[$contentId] = $responseBody;
             }
         } else {
             throw new \Exception('Unable to parse response.');
         }
 
+        if (!empty($failedRequests)) {
+            \Emergence\Mailer\Mailer::send('nafis@jarv.us', "Failed Responses (#$retry)", join('', ['<pre>', print_r($failedResponses, true), '</pre>']));
```

Comment:
Use the general logger instead, and in config files on your site you can change what log levels get emailed and where they get emailed